vault/external_tests/misc/kvv2_upgrade_test.go

```diff
@@ -18,10 +20,141 @@ import (
 	"github.com/kr/pretty"
 )
 
+type Buffer struct {
```

Comment:
I think specifying a Mutex in https://godoc.org/github.com/hashicorp/go-hclog#LoggerOptions might remove the need for this.